boto/utils.py

```diff
@@ -1049,3 +1050,39 @@ class RequestHook(object):
     """
     def handle_request_data(self, request, response, error=False):
         pass
+
+
+def host_is_ipv6(hostname):
+    """
+    Detect if the hostname is an IPV6 host using the socket module. Return
+    a boolean.
+    """
+    # empty strings or anything that is not a string is automatically not an
+    # IPV6 address
+    if not hostname or not isinstance(hostname, str):
+        return False
+
+    if hostname.startswith('['):  # assume we need to split on possible port
+        hostname = hostname.split(']:', 1)[0]
+    # strip leading/trailing brackets so inet_pton understands the hostname
+    hostname = hostname.strip('[]')
+    try:
+        socket.inet_pton(socket.AF_INET6, hostname)
```

Comment:
I'm not sure we should use this here-- support for this on Windows was only added in Python 3.4: https://docs.python.org/3.4/library/socket.html#socket.inet_pton